core/src/oauth/connection.rs

```diff
@@ -113,23 +133,23 @@ impl Connection {
         provider: ConnectionProvider,
         status: ConnectionStatus,
         metadata: serde_json::Value,
-        authorization_code: Option<String>,
-        access_token: Option<String>,
         access_token_expiry: Option<u64>,
-        refresh_token: Option<String>,
-        raw_json: Option<serde_json::Value>,
+        encrypted_authorization_code: Option<Vec<u8>>,
+        encrypted_access_token: Option<Vec<u8>>,
+        encrypted_refresh_token: Option<Vec<u8>>,
+        encrypted_raw_json: Option<Vec<u8>>,
     ) -> Self {
         Connection {
             connection_id,
             created,
             provider,
             status,
             metadata,
-            authorization_code,
-            access_token,
             access_token_expiry,
-            refresh_token,
-            raw_json,
+            encrypted_authorization_code,
```

Comment:
Just wondering if we need to encrypt all of those fields? I mean `access_token` and `refresh_token` obviously but the other 🤔.

Do you have an example of what `raw_json` contains from your test?

Reply:
Can be seen in nango.dev. It contains the access token and refresh token. We could store a sanitized version but that introduces more logic that can break with time. So instead the approach is to store it encrypted and expose a whiltelist of fields on the Connection (user related in general)

```
{
    "ok": true,
    "app_id": "A055TCFUZND",
    "authed_user": {
        "id": "U04VCD0FRKR"
    },
    "scope": "app_mentions:read,channels:history,channels:join,channels:read,chat:write,groups:read,im:history,im:read,mpim:read,team:read,users:read,users:read.email,metadata.message:read",
    "token_type": "bot",
    "access_token": "xoxb-...",
    "bot_user_id": "U055TFM6BU1",
    "team": {
        "id": "T050RH73H9P",
        "name": "Dust"
    },
    "enterprise": null,
    "is_enterprise_install": false
}
```